src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp

```diff
@@ -569,6 +569,14 @@ void ShenandoahConcurrentGC::op_final_mark() {
       }
     }
   }
+  EventGCHeapSummary event;
+  if (event.should_commit()) {
+    event.set_gcId(GCId::current());
+    event.set_when(GCWhen::AfterGC);
+    event.set_heapUsed(heap->used());
+    event.set_heapLive(heap->live());
+    event.commit();
+  }
```

Comment:
On the first sight, this belongs in `ShenandoahConcurrentMark::finish_mark()`. Placing the event here would fire the event when concurrent GC is cancelled, which is not what you want.

Reply:
:+1:

Reply:
Actually, this shouldn't even be here. `EventGCHeapSummary` is emitted via `trace_heap*` calls which should already be hooked into Shenandoah. Let me remove this.